worlds/doom_1993/__init__.py

```diff
@@ -0,0 +1,287 @@
+                   item["name"] != "Halls of the Damned (E2M6) - Yellow skull key":
+                    self.multiworld.itempool += [doom_1993_item]
+                    self.item_count += 1
+
+        # Place end level items in locked locations
+        for event in Events.events:
+            if not event in self.location_name_to_id:
+                continue
+            loc = Locations.location_table[self.location_name_to_id[event]]
+            if not self.included_episodes[loc["episode"] - 1]:
+                continue
+            self.multiworld.get_location(event, self.player).place_locked_item(
+                DOOM1993Item(event, ItemClassification.progression, None, player=self.player))
+            self.location_count -= 1
+    
+        # Special case for E2M6 and E3M8, where you enter a normal door then get stuck behind with a key door.
+        # We need to put the key in the locations available behind this door.
+        if self.included_episodes[1]:
+            self.place_locked_item_in_locations("Halls of the Damned (E2M6) - Yellow skull key", [
+                "Halls of the Damned (E2M6) - Yellow skull key",
+                "Halls of the Damned (E2M6) - Partial invisibility 2"
+            ])
+        if self.included_episodes[2]:
+            self.place_locked_item_in_locations("Warrens (E3M9) - Blue skull key", [
+                "Warrens (E3M9) - Rocket launcher",
+                "Warrens (E3M9) - Rocket launcher 2",
+                "Warrens (E3M9) - Partial invisibility",
+                "Warrens (E3M9) - Invulnerability",
+                "Warrens (E3M9) - Supercharge",
+                "Warrens (E3M9) - Berserk",
+                "Warrens (E3M9) - Chaingun"
+            ])
+
+        # Give starting levels right away
+        for i in range(len(self.included_episodes)):
+            if self.included_episodes[i]:
+                self.multiworld.push_precollected(self.create_item(self.starting_level_for_episode[i]))
+
+        # Fill the rest starting with weapons, powerups then fillers
+        self.create_ratioed_item("Armor")
+        self.create_ratioed_item("Mega Armor")
+        self.create_ratioed_item("Berserk")
+        self.create_ratioed_item("Invulnerability")
+        self.create_ratioed_item("Partial invisibility")
+        self.create_ratioed_item("Supercharge")
+
+        remaining_loc = self.location_count - self.item_count
+        fillers = [
+            "Medikit",
+            "Box of bullets",
+            "Box of rockets",
+            "Box of shotgun shells",
+            "Energy cell pack"
+        ]
+        while remaining_loc > 0:
+            remaining_loc -= 1
+            item = self.create_item_with_classification(self.multiworld.random.choice(fillers), ItemClassification.filler)
+            self.multiworld.itempool.append(item)
+            self.item_count += 1
+
+
+    def create_ratioed_item(self, item_name):
```

Comment:
Nitpick: Probably should be `create_ratioed_items` to get across that it makes multiple items.